pyanaconda/modules/localization/installation.py

```diff
@@ -58,3 +70,139 @@ def _write_language_configuration(self, lang, root):
         except IOError as ioerr:
             msg = "Cannot write language configuration file: {}".format(ioerr.strerror)
             raise LanguageInstallationError(msg)
+
+
+class KeyboardInstallationTask(Task):
+    """Installation task for the keyboard configuration."""
+
+    X_CONF_DIR = "/etc/X11/xorg.conf.d"
+    X_CONF_FILE_NAME = "00-keyboard.conf"
+    VC_CONF_FILE_PATH = "/etc/vconsole.conf"
+
+    def __init__(self, sysroot, x_layouts, switch_options, vc_keymap):
+        """Create a new task,
+
+        :param sysroot: a path to the root of the installed system
+        :type sysroot: str
+        :param x_layouts: list of x layout specifications
+        :type x_layouts: list(str)
+        :param switch_options: list of options for layout switching
+        :type switch_options: list(str)
+        :param vc_keymap: virtual console keyboard mapping name
+        :type vc_keymap: str
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._x_layouts = x_layouts
+        self._switch_options = switch_options
+        self._vc_keymap = vc_keymap
+
+    @property
+    def name(self):
+        return "Configure keyboard"
+
+    def run(self):
+        if self._x_layouts:
+            write_x_configuration(
+                self._x_layouts,
+                self._switch_options,
+                self._sysroot
+            )
+        if self._vc_keymap:
+            write_vc_configuration(
+                self._vc_keymap,
+                self._sysroot
+            )
+
+
+def write_x_configuration(x_layouts, switch_options, root):
+    """Write X keyboard layout configuration to the configuration files.
+
+    :param x_layouts: list of x layout specifications
+    :type x_layouts: list(str)
+    :param switch_options: list of options for layout switching
+    :type switch_options: list(str)
+    :param root: root path of the configured system
+    :type root: str
+    """
+
+    localed_wrapper = LocaledWrapper()
+    xconf_file_path = os.path.normpath(X_CONF_DIR + "/" + X_CONF_FILE_NAME)
+    errors = []
+
+    try:
+        if not os.path.isdir(X_CONF_DIR):
+            os.makedirs(X_CONF_DIR)
+    except OSError:
+        errors.append("Cannot create directory {}".format(X_CONF_DIR))
+
+    if root != "/":
```

Comment:
These large ``if-else`` should be abstracted, thought.

Reply:
It is not so easy.
This belongs to the point
- perhaps polish moved content

from my initial comment.

I am starting to thing that the two usages of the functions (runtime, chroot) should use separate functions, they share only a little of code and behave quite differently (eg root="/" version is using default). I may come with follow-up patch for this, maybe in scope of this PR, maybe not because I didn't want to introduce any non-necessary bugs/regressions caused by refactoring while moving (this PR).

Reply:
agree, lets go with the current solution for this PR